### PR TITLE
Add extensions array to configuration object

### PR DESCRIPTION
**Pull request by teles**
This pull request intends to allow custom output files, more information can be found in #18

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -14,6 +14,7 @@ import fs from 'fs'
  * @param {String} [config.iife] Optional. Specify filename to write Self executing <script>, exposing all icons as symbols on page,
  * @param {String} [config.dts] Optional. Specify filename to write TypeScript definitions with `export declare const`
  * @param {String} [config.dtsx] Optional. Specify filename to write TypeScript definitions for JSX with `export declare const`
+ * @param {includeBanner: Boolean, parser: Function, filename: String}[]} extensions Optional. List of extensions to generate custom outputs
```
Comment (skjalgepalg) [inline lib/index.js:17]: Missing a bracket and syntax to be part of config. Should also wrap object in array notation for easier legibility and put optional key last.
I suggest an update to the following: `@param {[{parser: Function, filename: String, includeBanner?: Boolean}]} [config.extensions]`

**Review by skjalgepalg — CHANGES_REQUESTED** [review]
This looks like a handy addition to the package, thanks for making the PR!

Added some comments where I think we should make some changes before merging.

File: lib/index.js [hunk 2]
```diff
@@ -58,7 +59,18 @@ export default function svgToJS (config) {
     esmx: `import React from 'react'${icons.map(({ titleCase, jsx }) => `\nexport function ${titleCase} (props) {${jsx}}`).join('')}`,
     dts: icons.map(({ camelCase }) => `export declare const ${camelCase}: string`).join('\n'),
     dtsx: icons.map(({ titleCase }) => `export declare const ${titleCase}: React.FunctionComponent<React.SVGProps<SVGElement>>`).join('\n')
-  }
+  };
+
+  // Merge extensions to result if extensions are specified in config
+  [].concat((config.extensions || [])
```
Comment (skjalgepalg) [inline lib/index.js:65]: Given the context (in config), `extensions` does feel a bit too generic and unclear; I suggest renaming to make the end result more evident. Something in the line of `customOutputs` feels more to the point.

File: README.md [hunk 3]
```diff
@@ -42,3 +42,58 @@ svgtojs(options)
 //  dtsx: String
 //}
 ```
+
+## Extensions
```
[the remark this answers is not in the document]
Comment (teles) [reply README.md:46]: Could we use "Using custom outputs" and replace all extensions with "Custom outputs" as you suggested ?

File: lib/index.js (hunk 1 above)
Comment (teles) [reply lib/index.js:17]: Thank you! JSDoc is a new stuff for me

**Comment by teles** [comment]
Hi, @skjalgepalg i made the suggested changes. 
Thank you for review my PR :) 
I agree that it would be better to call the extensions as custom outputs
Please let me know if any more change is needed

**Comment by teles** [comment]
Hi, @skjalgepalg. I guess I fixed all mentioned points
Thank you for the appointments
Please let me know if you find anything else

File: test/index.test.js [hunk 4]
```diff
@@ -57,3 +57,83 @@ describe('svg-to-js', () => {
...
+    expect(resultWithCustom.custom_2.includes(BANNER_TEXT)).toBe(true)
+  })
+
+  it('incomplete customOutputs should not be counted', () => {
+    const customEntries = Object.keys(resultWithCustom).filter(key => key.startsWith('custom_'))
+    const validcustomOutputs = customOutputs.filter(item => item.filename && item.parser)
+    expect(customEntries.length).toBe(validcustomOutputs.length)
+  })
+
+  it('includeBanners works as expected', () => {
```
[the remark this answers is not in the document]
Comment (teles) [reply test/index.test.js:126]: Nice suggestion, I forgot to clean up the files

**Review by skjalgepalg — APPROVED** [review]

**Comment by teles** [comment]
Hi, @eirikbacker @skjalgepalg! 
Is there anything else you would recommend me to do before having this PR merged ?

**Review by pingrid — APPROVED** [review]

**Pull request merged**